eis_toolkit/vector_processing/extract_shared_lines.py

```diff
@@ -0,0 +1,49 @@
+import geopandas as gpd
+from beartype import beartype
+
+from eis_toolkit import exceptions
+
+
+def _extract_shared_lines(polygons: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+    shared_lines_list = []
```

Comment:
This _could_ be defined as `shared_lines_list: Sequence = []`

Reply:
No need to define types for variables, except in maybe some rare cases.